RadixWallet/Clients/ResetWalletClient/ResetWalletClient+Live.swift

```diff
@@ -4,16 +4,24 @@ extension ResetWalletClient: DependencyKey {
 	public static let liveValue: Self = {
 		let walletDidResetSubject = AsyncPassthroughSubject<Void>()
 
+		@Dependency(\.errorQueue) var errorQueue
+		@Dependency(\.appPreferencesClient) var appPreferencesClient
 		@Dependency(\.cacheClient) var cacheClient
 		@Dependency(\.radixConnectClient) var radixConnectClient
 		@Dependency(\.userDefaults) var userDefaults
 
 		return Self(
 			resetWallet: {
-				cacheClient.removeAll()
-				await radixConnectClient.disconnectAll()
-				userDefaults.removeAll()
-				walletDidResetSubject.send(())
+				do {
+					try await appPreferencesClient.deleteProfileAndFactorSources(true)
```

Comment:
Why was this added?

Reply:
It was just moved. Previously, this was called in the Main reducer, after `walletDidResetSubject` fired. So this change is just about:
1. Collecting all the logic for resetting in the client
2. Perform that throwing task first, so we don't end up clearing cache and userDefaults, and failing to delete the Profile.